reconcile/openshift_saas_deploy_trigger_configs.py

```diff
@@ -4,38 +4,34 @@
 
 
 import reconcile.queries as queries
-import reconcile.jenkins_plugins as jenkins_base
 import reconcile.openshift_saas_deploy_trigger_base as osdt_base
 
+from reconcile.utils.defer import defer
 from reconcile.utils.semver_helper import make_semver
-from reconcile.utils.gitlab_api import GitLabApi
-from reconcile.utils.saasherder import SaasHerder
 
 
 QONTRACT_INTEGRATION = 'openshift-saas-deploy-trigger-configs'
 QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
 
 
-def run(dry_run, thread_pool_size=10):
-    saas_files = queries.get_saas_files()
+@defer
+def run(dry_run, thread_pool_size=10, internal=None,
+        use_jump_host=True, defer=None):
+    saas_files = queries.get_saas_files(v1=True, v2=True)
     if not saas_files:
         logging.error('no saas files found')
         sys.exit(1)
 
-    instance = queries.get_gitlab_instance()
-    settings = queries.get_app_interface_settings()
-    accounts = queries.get_aws_accounts()
-    gl = GitLabApi(instance, settings=settings)
-    jenkins_map = jenkins_base.get_jenkins_map()
-
-    saasherder = SaasHerder(
-        saas_files,
-        thread_pool_size=thread_pool_size,
-        gitlab=gl,
-        integration=QONTRACT_INTEGRATION,
-        integration_version=QONTRACT_INTEGRATION_VERSION,
-        settings=settings,
-        accounts=accounts)
+    setup_options = {
+        'saas_files': saas_files,
+        'thread_pool_size': thread_pool_size,
+        'internal': internal,
+        'use_jump_host': use_jump_host,
+        'integration': QONTRACT_INTEGRATION,
+        'integration_version': QONTRACT_INTEGRATION_VERSION,
+    }
+    saasherder, jenkins_map, oc_map, settings = osdt_base.setup(setup_options)
+    defer(lambda: oc_map.cleanup())
 
     trigger_specs = saasherder.get_configs_diff()
     already_triggered = []
```

Comment:
why do we need to pass this empty list?

Reply:
because we are populating it inside the `trigger` function and we need it consistent across all iterations:
https://github.com/app-sre/qontract-reconcile/pull/1579/files#diff-74302d8da8dcc6352d3477559844463eb65a0378418413759a1494be59621871R42

Reply:
the link points to this line:
```
for job_spec in trigger_specs:
```